tensorboard/tools/docs_list_format_test.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+# Check that any Markdown lists in IPython notebooks are preceded by a
+# blank line, which is required for the Google-internal docs processor
+# to interpret them as lists.
+
+set -e
+
+if ! [ -f WORKSPACE ]; then
+    printf >&2 'fatal: no WORKSPACE file found (are you at TensorBoard root?)\n'
+    exit 2
+fi
+
+git ls-files -z '*.ipynb' |
+xargs -0 awk '
+    { is_list = /"([-*]|[0-9]+\.) / }
```

Comment:
Maybe the trick is hidden in the `"` but I presume this cannot catch the case like below, right?
```md
   * foo\n
* bar\n
\n
```

Reply:
Correct, it won’t catch that. This is all a big approximation that is
neither sound nor complete (grepping over JSON strings representing
Markdown text in a serialized notebook file…) so I’m not *too* worried
about it being imperfect. If it becomes problematic, we can patch the
script or ditch it completely, but I figure that it’s better than
nothing for now.

Reply:
:+1: iterates the best :)